src/persist-client/examples/source_example.rs

```diff
@@ -1288,29 +1286,24 @@ mod render {
             let target_upper = &mut target_upper;
             let initial_upper = &mut initial_upper;
 
-            let mut input = (&mut write, &stashed_batches);
-
             println!("instance {}: initial_upper: {:?}", name, initial_upper);
-            let result = retry_mut("writer_append", &mut input, |(write, stashed_batches)| {
-                let target_upper1 = target_upper.clone();
-                let updates = stashed_batches
-                    .iter()
-                    .filter(move |batch| {
-                        !timely::PartialOrder::less_equal(&target_upper1, &batch.inclusive_upper)
-                    })
-                    .flat_map(|batch| batch.updates.iter());
-
-                // Filter out updates that have already been written by someone.
-                let initial_upper1 = initial_upper.clone();
-                let updates = updates
-                    .filter(move |(_update, ts)| initial_upper1.less_equal(ts))
-                    .map(|(update, ts)| ((update, ()), ts, 1));
-
-                let fut = write.append(updates, write.upper().clone(), target_upper.clone());
-
-                Box::pin(fut)
-            })
-            .await;
+            let target_upper1 = target_upper.clone();
+            let updates = stashed_batches
+                .iter()
+                .filter(move |batch| {
+                    !timely::PartialOrder::less_equal(&target_upper1, &batch.inclusive_upper)
+                })
+                .flat_map(|batch| batch.updates.iter());
+
+            // Filter out updates that have already been written by someone.
+            let initial_upper1 = initial_upper.clone();
+            let updates = updates
+                .filter(move |(_update, ts)| initial_upper1.less_equal(ts))
+                .map(|(update, ts)| ((update, ()), ts, 1));
+
+            let result = write
+                .append(updates, write.upper().clone(), target_upper.clone())
```

Comment:
@aljoscha this change causes a compile error about FnOnce lifetimes. any idea what's going on and/or how to fix it?

```
error[E0308]: mismatched types
    --> src/persist-client/examples/source_example.rs:1031:29
     |
1031 |         let consumer_task = mz_ore::task::spawn(|| "source-emit", async move {
     |                             ^^^^^^^^^^^^^^^^^^^ lifetime mismatch
     |
     = note: expected struct `std::slice::Iter<'_, _>`
                found struct `std::slice::Iter<'_, _>`
note: the lifetime requirement is introduced here
    --> /Users/dan/mz/materialize/src/ore/src/task.rs:54:19
     |
54   |     Fut: Future + Send + 'static,
     |                   ^^^^

error: implementation of `FnOnce` is not general enough
    --> src/persist-client/examples/source_example.rs:1031:29
     |
1031 |         let consumer_task = mz_ore::task::spawn(|| "source-emit", async move {
     |                             ^^^^^^^^^^^^^^^^^^^ implementation of `FnOnce` is not general enough
     |
     = note: closure with signature `fn(&'0 TimestampedBatch<std::string::String, <TS as Timestamper>::ToTimestamp>) -> std::slice::Iter<'_, (std::string::String, <TS as Timestamper>::ToTimestamp)>` must implement `FnOnce<(&TimestampedBatch<std::string::String, <TS as Timestamper>::ToTimestamp>,)>`, for any lifetime `'0`...
     = note: ...but it actually implements `FnOnce<(&TimestampedBatch<std::string::String, <TS as Timestamper>::ToTimestamp>,)>`

error: implementation of `FnOnce` is not general enough
    --> src/persist-client/examples/source_example.rs:1031:29
     |
1031 |         let consumer_task = mz_ore::task::spawn(|| "source-emit", async move {
     |                             ^^^^^^^^^^^^^^^^^^^ implementation of `FnOnce` is not general enough
     |
     = note: closure with signature `fn(&'0 (std::string::String, <TS as Timestamper>::ToTimestamp)) -> ((&std::string::String, ()), &<TS as Timestamper>::ToTimestamp, i64)` must implement `FnOnce<(&(std::string::String, <TS as Timestamper>::ToTimestamp),)>`, for any lifetime `'0`...
     = note: ...but it actually implements `FnOnce<(&(std::string::String, <TS as Timestamper>::ToTimestamp),)>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `mz-persist-client` due to 3 previous errors
```